--- origsrc/gwibber-2.31.95/gwibber/microblog/dispatcher.py	2010-09-22 11:47:01.000000000 -0500
+++ src/gwibber-2.31.95/gwibber/microblog/dispatcher.py	2010-09-26 00:57:33.344914700 -0500
@@ -610,8 +610,8 @@ class ConnectionMonitor(dbus.service.Obj
     bus_name = dbus.service.BusName("com.Gwibber.Connection", bus=self.bus)
     dbus.service.Object.__init__(self, bus_name, self.__dbus_object_path__)
 
-    self.sysbus = dbus.SystemBus()
     try:
+      self.sysbus = dbus.SystemBus()
       self.nm = self.sysbus.get_object(NM_DBUS_SERVICE, NM_DBUS_OBJECT_PATH)
       self.nm.connect_to_signal("StateChanged", self.on_connection_changed)
     except:
--- origsrc/gwibber-2.31.95/gwibber/microblog/util/keyring.py	2010-09-16 23:22:53.000000000 -0500
+++ src/gwibber-2.31.95/gwibber/microblog/util/keyring.py	2010-09-26 00:06:57.819618900 -0500
@@ -1,6 +1,6 @@
 from const import *
 
-import ctypes
+import ctypes, ctypes.util
 import gnomekeyring
 import exceptions
 
@@ -17,7 +17,7 @@ def get_secret(id):
 
   return value
 
-libc = ctypes.CDLL("libc.so.6")
+libc = ctypes.CDLL(ctypes.util.find_library("c"))
 
 def mlock(var):
     libc.mlock(var, len(var))
